src/Init/Meta.lean

```diff
@@ -234,6 +235,47 @@ instance monadNameGeneratorLift (m n : Type → Type) [MonadLift m n] [MonadName
   setNGen := fun ngen => liftM (setNGen ngen : m _)
 }
 
+namespace TSyntax
+
+instance : Coe (TSyntax [k]) (TSyntax (k :: ks)) where
+  coe stx := ⟨stx⟩
+
+instance [Coe (TSyntax [k]) (TSyntax ks)] : Coe (TSyntax [k]) (TSyntax (k' :: ks)) where
+  coe stx := ⟨stx⟩
+
+instance : Coe (TSyntax identKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : CoeDep (TSyntax `term) ⟨Syntax.ident info ss n res⟩ (TSyntax `ident) where
+  coe := ⟨Syntax.ident info ss n res⟩
+
+instance : Coe (TSyntax strLitKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
```

Comment:
Another option to avoid the repetition here would be to introduce an auxiliary type class:
```lean
class SyntaxSubset (k l : Name) : Prop

instance [SyntaxSubset k l] : Coe (TSyntax [k]) (TSyntax [l]) where
  coe s := ⟨s.raw⟩

instance : SyntaxSubset strLitKind `term := ⟨⟩
instance : SyntaxSubset nameLitKind `term := ⟨⟩
-- etc.
```

This also has the nice side-effect of documenting the inclusion relationships between the "kinds".

Reply:
I suppose the question is, is `ident` a subset of `declId` even if they are structurally different? Or should we have both the subset coercion and "other" coercions?

Reply:
Ah, I didn't see of the `declId` coercion.  That's pretty useful!

Reply:
> Or should we have both the subset coercion and "other" coercions?

I would say both. Subset coercions for categories and other coercions for cases like the one you mentioned. I think the benefit @gebner mentioned of documenting category inclusion relationships in the type system is worth it.

Reply:
Since there is no direct need for `SyntaxSubset`, the declarations are now much shorter due to the aliases, and `SyntaxSubset` instances cannot directly be made safe using `Unhygienic` quotations, I'll consider this resolved for now.